x/gamm/keeper/pool_service.go

```diff
@@ -200,7 +200,7 @@ func (k Keeper) JoinPoolNoSwap(
 	// check that needed lp liquidity does not exceed the given `tokenInMaxs` parameter. Return error if so.
 	// if tokenInMaxs == 0, don't do this check.
 	if tokenInMaxs.Len() != 0 {
-		if !(neededLpLiquidity.DenomsSubsetOf(tokenInMaxs) && tokenInMaxs.IsAllGTE(neededLpLiquidity)) {
+		if !(neededLpLiquidity.DenomsSubsetOf(tokenInMaxs) && tokenInMaxs.IsAllGTE(neededLpLiquidity) && tokenInMaxs.DenomsSubsetOf(neededLpLiquidity)) {
```

Comment:
Also, please add a test case for the new condition

Reply:
MB – added!